### PR TITLE
switch ordering of logical and

### LINKED ISSUE #11460: Check .length before accessing characters
See comment here: https://github.com/facebook/react/pull/11457#discussion_r148954193.

Although it seems very unlikely that someone would pass a one-character prop, the price (at least in versions of V8 in Chrome right now) is reported to be about 10x.

So we should check `.length` first and then do the comparison.

**Please comment if you want to take this issue.**
**If somebody already commented, please let them finish it.**

### PR DESCRIPTION
**Pull request by xjlim**
Fix #11460

### CODE REVIEW

**Comment by gaearon** [comment]
I think this is good. Let's wait for CI to pass.

**Comment by xjlim** [comment]
@gaearon I changed [this](https://github.com/facebook/react/blob/master/packages/react-dom/src/shared/ReactDOMUnknownPropertyHook.js#L71) too as they seem to follow the same behaviour.

File: packages/react-dom/src/shared/ReactDOMUnknownPropertyHook.js [hunk 1]
```diff
@@ -68,7 +68,7 @@ if (__DEV__) {
       return true;
     }
 
-    if (lowerCasedName.indexOf('on') === 0 && lowerCasedName.length > 2) {
+    if (lowerCasedName.length > 2 && lowerCasedName.indexOf('on') === 0) {
```
Comment (gaearon) [inline packages/react-dom/src/shared/ReactDOMUnknownPropertyHook.js]: This one is not necessary. We're guarding against `[]` access outside of bounds in particular. Here it doesn't happen.

Comment (xjlim) [reply packages/react-dom/src/shared/ReactDOMUnknownPropertyHook.js]: I see. Sorry I accidentally messes up with the git, should I squash them?

Comment (gaearon) [reply packages/react-dom/src/shared/ReactDOMUnknownPropertyHook.js]: Doesn't matter, we squash on merge anyway.

**Pull request merged**